### DISCUSSION TITLE
Directus is making a request for user permissions with an empty object

### DISCUSSION THREAD

**Discussion by Industrial** [directus/directus#14708] [Archive]
### Describe the Bug

Hi.

I am trying to mount Directus as an API route within Next.js. This allows it to be deployed to lambda/serverless/function/edge environments like Vercel, AWS, etc.

See https://github.com/Industrial/test-next-directus for the repository. This is how I'm mounting the Express app: https://github.com/Industrial/test-next-directus/blob/main/pages/api/%5B...cms%5D.ts

When I start the Next.js server and go to `http://localhost:3000/api/cms` it correctly redirects to `http://localhost:3000/api/cms/admin` and then shows me the login page. I can login with the admin user (SQLite).

When the request is made for the user info, the response contains the user's data including the role.
Directus then makes this request (which fails): http://localhost:3000/api/cms/permissions?limit=-1&filter={"role":{}}

I'm not sure where to go from here. It looks like all the core functionality in Directus should be working because the Express App is mounted and is serving routes.

### To Reproduce

```bash
git clone https://github.com/Industrial/test-next-directus.git
cd test-next-directus
pnpm i
pnpm dev
```

* Go to http://localhost:3000/api/cms
* Open the inspector (Ctrl + Shift + I)
* Login with user `admin@example.com` and password `admin`.
* Observe the requests in the network tab

### Errors Shown

{"level":50,"time":1658955341035,"pid":23711,"hostname":"Monster","extensions":{"stack":"TypeError: Cannot convert undefined or null to object\n    at Function.keys (<anonymous>)\n    at getFilterPath (/home/tom/Code/code9/node/test-next-directus/node_modules/.pnpm/registry.npmjs.org+directus@9.14.5_xrxvbtylmve4l2tr3vmmqgfp7q/node_modules/directus/dist/utils/apply-query.js:478:23)\n    at getFilterPath (/home/tom/Code/code9/node/test-next-directus/node_modules/.pnpm/registry.npmjs.org+directus@9.14.5_xrxvbtylmve4l2tr3vmmqgfp7q/node_modules/directus/dist/utils/apply-query.js:482:22)\n    at addJoins (/home/tom/Code/code9/node/test-next-directus/node_modules/.pnpm/registry.npmjs.org+directus@9.14.5_xrxvbtylmve4l2tr3vmmqgfp7q/node_modules/directus/dist/utils/apply-query.js:170:32)\n    at applyFilter (/home/tom/Code/code9/node/test-next-directus/node_modules/.pnpm/registry.npmjs.org+directus@9.14.5_xrxvbtylmve4l2tr3vmmqgfp7q/node_modules/directus/dist/utils/apply-query.js:155:5)\n    at applyQuery (/home/tom/Code/code9/node/test-next-directus/node_modules/.pnpm/registry.npmjs.org+directus@9.14.5_xrxvbtylmve4l2tr3vmmqgfp7q/node_modules/directus/dist/utils/apply-query.js:43:9)\n    at getDBQuery (/home/tom/Code/code9/node/test-next-directus/node_modules/.pnpm/registry.npmjs.org+directus@9.14.5_xrxvbtylmve4l2tr3vmmqgfp7q/node_modules/directus/dist/database/run-ast.js:161:38)\n    at run (/home/tom/Code/code9/node/test-next-directus/node_modules/.pnpm/registry.npmjs.org+directus@9.14.5_xrxvbtylmve4l2tr3vmmqgfp7q/node_modules/directus/dist/database/run-ast.js:36:25)\n    at async runAST (/home/tom/Code/code9/node/test-next-directus/node_modules/.pnpm/registry.npmjs.org+directus@9.14.5_xrxvbtylmve4l2tr3vmmqgfp7q/node_modules/directus/dist/database/run-ast.js:30:16)\n    at async PermissionsService.readByQuery (/home/tom/Code/code9/node/test-next-directus/node_modules/.pnpm/registry.npmjs.org+directus@9.14.5_xrxvbtylmve4l2tr3vmmqgfp7q/node_modules/directus/dist/services/items.js:216:25)"},"stack":"TypeError: Cannot convert undefined or null to object\n    at Function.keys (<anonymous>)\n    at getFilterPath (/home/tom/Code/code9/node/test-next-directus/node_modules/.pnpm/registry.npmjs.org+directus@9.14.5_xrxvbtylmve4l2tr3vmmqgfp7q/node_modules/directus/dist/utils/apply-query.js:478:23)\n    at getFilterPath (/home/tom/Code/code9/node/test-next-directus/node_modules/.pnpm/registry.npmjs.org+directus@9.14.5_xrxvbtylmve4l2tr3vmmqgfp7q/node_modules/directus/dist/utils/apply-query.js:482:22)\n    at addJoins (/home/tom/Code/code9/node/test-next-directus/node_modules/.pnpm/registry.npmjs.org+directus@9.14.5_xrxvbtylmve4l2tr3vmmqgfp7q/node_modules/directus/dist/utils/apply-query.js:170:32)\n    at applyFilter (/home/tom/Code/code9/node/test-next-directus/node_modules/.pnpm/registry.npmjs.org+directus@9.14.5_xrxvbtylmve4l2tr3vmmqgfp7q/node_modules/directus/dist/utils/apply-query.js:155:5)\n    at applyQuery (/home/tom/Code/code9/node/test-next-directus/node_modules/.pnpm/registry.npmjs.org+directus@9.14.5_xrxvbtylmve4l2tr3vmmqgfp7q/node_modules/directus/dist/utils/apply-query.js:43:9)\n    at getDBQuery (/home/tom/Code/code9/node/test-next-directus/node_modules/.pnpm/registry.npmjs.org+directus@9.14.5_xrxvbtylmve4l2tr3vmmqgfp7q/node_modules/directus/dist/database/run-ast.js:161:38)\n    at run (/home/tom/Code/code9/node/test-next-directus/node_modules/.pnpm/registry.npmjs.org+directus@9.14.5_xrxvbtylmve4l2tr3vmmqgfp7q/node_modules/directus/dist/database/run-ast.js:36:25)\n    at async runAST (/home/tom/Code/code9/node/test-next-directus/node_modules/.pnpm/registry.npmjs.org+directus@9.14.5_xrxvbtylmve4l2tr3vmmqgfp7q/node_modules/directus/dist/database/run-ast.js:30:16)\n    at async PermissionsService.readByQuery (/home/tom/Code/code9/node/test-next-directus/node_modules/.pnpm/registry.npmjs.org+directus@9.14.5_xrxvbtylmve4l2tr3vmmqgfp7q/node_modules/directus/dist/services/items.js:216:25)","type":"Error","msg":"Cannot convert undefined or null to object"}

### What version of Directus are you using?

v9.14.5

### What version of Node.js are you using?

v18.4.0

### What database are you using?

SQlite

### What browser are you using?

Firefox

### How are you deploying Directus?

locally

**Comment by br41nslug** [comment]
I've converted this to a discussion as i doubt this is a supported use-case 😬 however this error is to be expected for an empty filter like `filter={"role":{}}` the request generated by the app *should* look like `/permissions?filter={"role":{"_null":true}}&limit=-1`

Im not sure what could be causing such a filter to be generated but my guess would be something to do with vue store persistence in the app.
